online-banking-cz/src/main/java/com/adyen/checkout/onlinebankingcz/OnlineBankingCZDelegate.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by atef on 25/8/2022.
+ */
+
+package com.adyen.checkout.onlinebankingcz
+
+import com.adyen.checkout.components.PaymentComponentState
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+import com.adyen.checkout.components.model.payments.request.OnlineBankingCZPaymentMethod
+import com.adyen.checkout.components.model.payments.request.PaymentComponentData
+import com.adyen.checkout.components.util.PaymentMethodTypes
+import com.adyen.checkout.issuerlist.IssuerListDelegate
+import com.adyen.checkout.issuerlist.IssuerListInputData
+import com.adyen.checkout.issuerlist.IssuerListOutputData
+import com.adyen.checkout.issuerlist.IssuerModel
+import com.adyen.checkout.issuerlist.IssuersUtils
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+
+class OnlineBankingCZDelegate(
+    private val paymentMethod: PaymentMethod,
+    private val onlineBankingCZPaymentMethod: () -> OnlineBankingCZPaymentMethod
```

Comment:
`paymentMethodFactory` would be a better name. Right now, I was confused why this would be a lambda, but after diving in the code I realised there is a good reason for it